### DISCUSSION TITLE
Stonecutter and Architectury version parsing

### DISCUSSION THREAD

**Discussion by FamroFexl** [Kira-NT/mc-publish#128] [Q&A] [answered]
I am using a Github release format where the tag is just a version number such as "0.1". The name of the release that uses that tag is called "0.1-alpha". In that version release, I need all my minecraft versions and loaders. For instance, if I was supporting 1.20 on forge and fabric, and 1.21 on just fabric, I would have three files in the "0.1-alpha" release called "modid-0.1-1.20-fabric", "modid-0.1-1.20-forge", and "modid-0.1-1.21-fabric". 

Would mc-publish be able to support a use case like this, where it publishes multiple files for different minecraft versions and mod loaders for a single mod version? If not, could I configure Github Actions to support this or would I just have to fork this repository and add my own features?

**Comment by Kira-NT** [comment, marked as answer]
> I am using a Github release format where the tag is just a version number such as "0.1".

`mc-publish` doesn't really care about your naming convention, as it reads the values it needs directly from the metadata of the mod you're trying to upload.

> Would mc-publish be able to support a use case like this, where it publishes multiple files for different minecraft versions and mod loaders for a single mod version?

You can achieve whatever you need with `mc-publish`, as every aspect of it is configurable. While I would advise against grouping different mods under a single version - since Modrinth/CurseForge would treat some of them as secondary files, reducing their visibility to end users - you can still upload multiple files as a single mod version. Just use the `files` input. You may also want to override the `loaders` input, as `mc-publish` will populate that field based on the primary file alone.

If you decide to treat your mods as separate versions *(which, to be honest, they are)*, you can refer to #29.
